core/src/worker/workflow/managed_run.rs

```diff
@@ -988,7 +988,8 @@ impl ManagedRun {
         let should_respond = !(self.wfm.machines.has_pending_jobs()
             || outgoing_cmds.replaying
             || is_query_playback
-            || data.activation_was_only_eviction);
+            || data.activation_was_only_eviction
+            || self.wfm.machines.have_seen_terminal_event);
```

Comment:
This was added because, now that replaying gets flipped to false upon seeing the workflow execution's terminal event, a couple unit tests that were asserting things about completions saw different results on those final completions.

Thing is, those completions are in "response" to impossible tasks from the server. Server will never send a WFT with a terminal event in it (except queries, which are permitted just below). So, now such completions are never "sent", which all probably makes more sense than it used to anyway.